src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -278,6 +281,10 @@ class Crystal::Doc::Generator
   end
 
   def compute_repository
+    # check whether inside git work-tree
+    `git rev-parse --is-inside-work-tree >/dev/null 2>&1`
```

Comment:
Is this additional command really necessary?

Reply:
The below `git remote -v` will output an error message without this. This is needed.